lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -205,13 +202,10 @@ def paths_to_3d_segments_with_codes(paths, zs=0, zdir='z'):
     """
 
     zs = np.broadcast_to(zs, len(paths))
-    segments = []
-    codes_list = []
-    for path, pathz in zip(paths, zs):
-        segs, codes = path_to_3d_segment_with_codes(path, pathz, zdir)
-        segments.append(segs)
-        codes_list.append(codes)
-    return segments, codes_list
+    segments, codes_list = zip(
+        *[path_to_3d_segment_with_codes(path, pathz, zdir)
+          for path, pathz in zip(paths, zs)])
```

Comment:
Related to https://github.com/matplotlib/matplotlib/pull/11577, this should be faster,
```py
In [4]: sequence = list(zip(range(1000), range(1000)))                                                                                                       

In [5]: %%timeit 
   ...: x = [] 
   ...: y = [] 
   ...: for x_el, y_el in sequence: 
   ...:     x.append(x_el) 
   ...:     y.append(y_el) 
   ...:                                                                                                                                                      
135 µs ± 498 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [6]: %%timeit 
   ...: x, y = zip(*sequence) 
   ...: x = list(x) 
   ...: y = list(y) 
   ...:  
   ...:                                                                                                                                                      
49.2 µs ± 234 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

Reply:
This is indeed a good performance improvement! However, it gets hard to read because there's too much going on. I propose to construct the list beforehand and give it a name. This should be equally fast.

~~~
    segments_and_codes_list = [
        path_to_3d_segment_with_codes(path, pathz, zdir)
        for path, pathz in zip(paths, zs)
    ]
    segments, codes = zip(*segments_and_codes_list)
    return list(segments), list(codes)
~~~